src/services/expenses/expenses.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const { v4: uuid } = require('uuid');
+
+const { db } = require('../../db');
+
+const { validateExpenseData } = require('./validation');
+
+const resetStore = async () => {
+  await db`DROP TABLE IF EXISTS expenses`;
+
+  await db`CREATE TABLE expenses (
+    id UUID PRIMARY KEY,
+    userId UUID,
+    title TEXT,
+    category TEXT,
+    amount NUMERIC,
+    spentAt TIMESTAMP,
+    note TEXT
+  )`;
+};
+
+const getAll = async ({ userId, categories, from, to }) => {
+  return await db`
+    SELECT *
+    FROM expenses
+    WHERE
+      userId = ${userId}
+      AND
+      spentAt BETWEEN ${from} AND ${to}
+      AND
+      category IN (${categories.join(', ')})
+  `;
+};
+
+const getById = async (id) => {
+  const response = await db`SELECT * FROM expenses WHERE id = ${id}`;
+
+  return response[0];
+};
+
+const create = async (expenseData) => {
+  validateExpenseData(expenseData);
+
+  const { userId, spentAt, title, amount, category, note } = expenseData;
+
+  const response = await db`
+    INSERT INTO expenses (id, userId, spentAt, title, amount, category, note)
+    VALUES (
+      ${uuid()},
+      ${userId},
+      ${spentAt},
+      ${title},
+      ${amount},
+      ${category},
+      ${note}
+    )
+    RETURNING *
+  `;
+
+  return response[0];
+};
+
+const deleteById = async (id) => {
+  const response = await db`
+    DELETE FROM expenses
+    WHERE id = ${id}
+    RETURNING id
+  `;
+
+  return response[0];
+};
+
+const update = async (id, updatedExpense) => {
+  const { title, amount, category, note, spentAt } = updatedExpense;
+
```

Comment:
there should be smth like that:
```
  const setProperties = Object.entries(updatedExpense)
    .filter(([_key, value]) => value !== undefined)
    .map(([key, value]) => `${key} = ${value}`)
    .join(', ');
  ```
  
 but it doesn't work due to the way `postgres` library parses the template (!?)
 even if 
 `setProperties === 'title = ${title},
      amount = ${amount},
      category = ${category},
      note = ${note},
      spentAt = ${spentAt}'`
      
   the same with filter options in `getAll` function (now it works if all params are defined)
   in the real project, I'd use smth like Sequelize or Prisma library but to get used to postgress query syntax and not make task solution more complex I leave `postgres.Sql`

Reply:
It's ok to have every field manually. Explicit over implicit.